backend/src/main/java/util/SystemCall.java

```diff
@@ -16,17 +21,33 @@ public SystemCallResult fire(ExecutorService executorService, String command) {
         try {
             Process process = builder.start();
 
-            StringBuilder stdOutStringBuilder = new StringBuilder();
-            StreamEater outStreamGobbler = new StreamEater(process.getInputStream(), e -> stdOutStringBuilder.append(e).append("\n"));
+            final StringBuilder stdOutStringBuilder = new StringBuilder();
+            final StreamEater outStreamGobbler = new StreamEater(process.getInputStream(), line -> stdOutStringBuilder.append(line).append("\n"));
             executorService.submit(outStreamGobbler);
 
-            StringBuilder stdErrStringBuilder = new StringBuilder();
-            StreamEater errStreamGobbler = new StreamEater(process.getErrorStream(), e -> stdErrStringBuilder.append(e).append("\n"));
+            final StringBuilder stdErrStringBuilder = new StringBuilder();
+            final StreamEater errStreamGobbler = new StreamEater(process.getErrorStream(), line -> stdErrStringBuilder.append(line).append("\n"));
             executorService.submit(errStreamGobbler);
 
-            return new SystemCallResult(process.waitFor(), stdOutStringBuilder.toString(), stdErrStringBuilder.toString());
+            SystemCallResult systemCallResult = new SystemCallResult(process.waitFor(),
+                    stdOutStringBuilder.toString(), stdErrStringBuilder.toString());
+
+            return checkExitCode(command, systemCallResult, expectedExitCode);
+
         } catch (IOException | InterruptedException exception) {
-            throw new RuntimeException(exception);
+            logger.severe(String.format("Failed running system call '%s': %s", command, exception.getMessage()));
+            return Optional.empty();
+        }
+    }
+
+    private Optional<SystemCallResult> checkExitCode(String command, SystemCallResult systemCallResult, int expectedExitCode) {
+        if (systemCallResult.getExitCode() == expectedExitCode) {
+            return Optional.of(systemCallResult);
+        } else {
+            final String errorMessage = String.format("Command '%s' returned with unexpected exit code '%d': '%s'",
+                    command, systemCallResult.getExitCode(), systemCallResult);
+            logger.severe(errorMessage);
+            return Optional.empty();
```

Comment:
Very Functional error handling here, might be confusing for plain old java developers but seems to do the job nicely for getting rid of that system call hassle! 👍

Reply:
I am not no sure if the functional style is right - or better: I don't know in which situation it is better to write code in functional style... ? 
Here it might be wrong - but I don't want to change everything again.
The goal is to get rid of this SystemCall monster - but I still have some problems with the mpd libray reading the current volume value from the player object.